server/channels/app/ldap.go

```diff
@@ -71,6 +71,25 @@ func (a *App) TestLdapConnection(rctx request.CTX, settings *model.LdapSettings)
 	return nil
 }
 
+func (a *App) TestLdapFilters(rctx request.CTX, settings *model.LdapSettings) ([]model.LdapFilterTestResult, *model.AppError) {
+	license := a.Srv().License()
+	ldapI := a.LdapDiagnostic()
+
+	// NOTE: normally we would test (*a.Config().LdapSettings.Enable || *a.Config().LdapSettings.EnableSync),
+	// but we want to allow sysadmins to test the connection without enabling and saving the config first.
+	if ldapI != nil && license != nil && *license.Features.LDAP {
+		res, err := ldapI.RunTestFilters(rctx, settings)
+		if err != nil {
+			err.StatusCode = 500
```

Comment:
Is this a debug leftover? We are already setting statusCode to 500 from the enterprise layer.

Reply:
Same thing -- LDAP is using this pattern. But I'll start changing it (at least the new code), then I'll go through and normalize the older code to current standards.